### DISCUSSION TITLE
How do queue slots work?

### DISCUSSION THREAD

**Discussion by Lord-Kamina** [gabime/spdlog#3199] [Q&A] [answered]
This is possibly a dumb question in that it arose from a literal reading of the wiki; specifically:

[`All the queue slots are pre-allocated on thread-pool construction (each slot occupies ~256 bytes on 64bit systems)`](https://github.com/gabime/spdlog/wiki/6.-Asynchronous-logging#spdlogs-thread-pool)

What I'm wondering is... what happens if a specific log entry is larger than that? Does the queue slot get resized, or does something else happen (i.e. the message gets split into more than one slot or truncated, or something like that)?

**Comment by gabime** [comment, marked as answer]
If specific message is bigger than 250 bytes, more bytes will be allocated by the buffer [member](https://github.com/gabime/spdlog/blob/e593f6695c6065e6b345fe2862f04a519ed484e0/include/spdlog/details/log_msg_buffer.h#L15) of the slot.

**Reply by Lord-Kamina** [reply]
Perfect, makes sense.
Thank you for always providing lightning-fast replies 😂
